scene/3d/collision_shape_3d.cpp

```diff
@@ -111,23 +111,34 @@ void CollisionShape3D::resource_changed(RES res) {
 }
 
 String CollisionShape3D::get_configuration_warning() const {
+	String warning = Node3D::get_configuration_warning();
+
 	if (!Object::cast_to<CollisionObject3D>(get_parent())) {
-		return TTR("CollisionShape3D only serves to provide a collision shape to a CollisionObject3D derived node. Please only use it as a child of Area3D, StaticBody3D, RigidBody3D, KinematicBody3D, etc. to give them a shape.");
+		if (!warning.empty()) {
+			warning += "\n\n";
+		}
+		return warning + TTR("CollisionShape3D only serves to provide a collision shape to a CollisionObject3D derived node. Please only use it as a child of Area3D, StaticBody3D, RigidBody3D, KinematicBody3D, etc. to give them a shape.");
```

Comment:
Same here about the early return, these checks are designed for the ability to have several warnings appended to `warning`, if there are several issues to warn about.